### PR TITLE
Feature/pat support

### PR DESCRIPTION
**Pull request by pascal-fischer**
**This PR extends the settings page with support for personal access tokens by splitting the settings page into personal settings and account settings.**

The personal settings page includes a tokens sections that lists all personal access tokens of a user. It allows to create new ones and delete existing ones.
When creating a token the user needs to provide a name and how many days the token should be valid for.
After the token is created the UI shows the generated token in plain text for the user to copy.

The link to the documentation is disabled until available.

Screenshot 2023-03-27 at 16 56 03

Screenshot 2023-03-27 at 16 56 10
Screenshot 2023-03-27 at 13 29 57
Screenshot 2023-03-27 at 13 27 36

### CODE REVIEW

File: src/views/Settings.tsx [hunk 1]
```diff
@@ -1,227 +1,48 @@
...
-                                                    disabled={!formPeerExpirationEnabled}
-                                                    options={Array.of({key: "hour", title: "Hours"}, {
-                                                        key: "day",
-                                                        title: "Days"
-                                                    })
-                                                    }/>
-                                            </Form.Item>
-                                            <Form.Item>
-                                                <Button icon={<QuestionCircleFilled/>} type="link" target="_blank"
-                                                        href="https://netbird.io/docs/how-to-guides/periodic-authentication">Learn more about login expiration</Button>
```
Comment (braginini) [inline src/views/Settings.tsx:208]: I think that it makes sense to move it more to the left

**Review by braginini — CHANGES_REQUESTED** [review]

File: src/views/SettingsPersonal.tsx [hunk 2]
```diff
@@ -0,0 +1,511 @@
...
+        }
+    }
+
+    return (
+        <>
+            <Container style={{paddingTop: "40px"}}>
+                <Row>
+                    <Col span={24}>
+                        <Title level={4}>Personal Access Tokens</Title>
+                        <Paragraph>A list of all the personal access tokens for your user.</Paragraph>
```
[the remark this answers is not in the document]
Comment (pascal-fischer) [reply src/views/SettingsPersonal.tsx:308]: On the addToken popup I have a link to the not yet existing documentation about PATs. I don't know if we also need an explanation on the normal settings page

File: src/views/SettingsPersonal.tsx [hunk 3]
```diff
@@ -0,0 +1,511 @@
...
+                        {showPlainToken ? "Token created successfully!" : "Create new Personal Access Token"}
+                    </Paragraph>
+                    <Paragraph type={"secondary"}
+                               style={{
+                                   textAlign: "center",
+                                   whiteSpace: "pre-line",
+                                   marginTop: "-25px",
+                                   paddingBottom: "35px",
+                               }}>
+                        {showPlainToken ? "You will only see this token once so copy it and store it in a secure location." : "This token can be used to authenticate against our API."}
```
[the remark this answers is not in the document]
Comment (pascal-fischer) [reply src/views/SettingsPersonal.tsx:435]: yes this should definitely be changed, will do that

File: src/views/SettingsPersonal.tsx [hunk 4]
```diff
@@ -0,0 +1,511 @@
...
+                                <Divider style={{marginTop: "0px"}}></Divider>
+                                <Button icon={<QuestionCircleFilled/>} type="link" target="_blank" disabled={true}
+                                        href="https://netbird.io/docs/overview/personal-access-tokens">Learn more about personal access tokens</Button>
+                            </Col>
+                        </Row>
+                    </Form>}
+                    {showPlainToken && <Space className="nb-code" direction="vertical" size="middle">
+                        <Row>
+                            <>
+                                <Text style={{fontSize: "large", marginTop: "-4px", backgroundColor: "#ebeef3", padding: "10px", borderRadius: "5px", marginBottom: "20px"}}>{plainToken}</Text>
```
Comment (braginini) [inline src/views/SettingsPersonal.tsx:493]: I think that it makes sense to use the same component - SyntaxHighlighter as in TabSteps.tsx

**Review by braginini — CHANGES_REQUESTED** [review]

File: src/views/Settings.tsx (hunk 1 above)
Comment (pascal-fischer) [reply src/views/Settings.tsx:208]: First that's a delete part. But I moved it someplace else. I can change the indention but that somehow breaks the level, I don't know if that is right, every code formatter will move it back again

File: src/views/SettingsPersonal.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (pascal-fischer) [reply src/views/SettingsPersonal.tsx]: Updated the explaination

File: src/views/SettingsPersonal.tsx (hunk 4 above)
Comment (pascal-fischer) [reply src/views/SettingsPersonal.tsx]: switched to SyntaxHighlighter

**Review by braginini — APPROVED** [review]

**Review by mlsmaycon — APPROVED** [review]

**Pull request merged**